pkg/envoy/xds/cache.go

```diff
@@ -215,6 +215,9 @@ func (c *Cache) GetResources(typeURL string, lastVersion uint64, nodeIP string,
 		res.Resources = make([]proto.Message, 0, len(c.resources))
 		cacheLog.Debugf("no resource names requested, returning all %d resources", len(c.resources))
 		for k, v := range c.resources {
+			if k.typeURL != typeURL {
```

Comment:
Is this a bug? Also, is this covered by a unit test?

If it is a valid change independent of this patch set, might want to consider merging this as a separate PR (with the test).

Reply:
the fix was merged to main and I rebased the code